crates/bevy_ecs/src/storage/blob_vec.rs

```diff
@@ -59,7 +59,7 @@ impl BlobVec {
     pub fn reserve(&mut self, amount: usize) {
         let available_space = self.capacity - self.len;
         if available_space < amount {
-            self.grow(amount - available_space);
+            self.grow(amount);
```

Comment:
This is not quite right. If the length is 2 and the capacity is 10, reserving 11 elements should give a capacity of 12, not 20. `Vec::reserve` grows the backing allocation enough to be able to push a certain amount of new elements. It doesn't make sense for `BlobVec` to do something different.

Reply:
Agreed. I think this is an issue with how Table tracks capacity / how it calls reserve on its Columns / Entities.

Reply:
I thought `BlobVec` was something specific to `Table` and as such didn't need to keep those counts, as it's only used within `Table`... but `BlobVec` is public while `Table` is not

Reply:
I think i have a solution. Creating a new pr now.